Purge the first edition stubs from search engines

kornelski opened the pull request:
Apparently this is still an annoying problem for users: https://users.rust-lang.org/t/why-are-links-to-the-old-book-still-floating-around/24656

This adds `<meta name="robots" content="noindex,follow">` to first edition's pages in order to remove them from search engines.

To do that I had to copy the whole index.hbs ~~and update move the theme directory to location required by the current mdBook version.~~

kornelski commented:
* [x] it has to work well with docs offline as well as online — it does, browsers ignore the robots meta.
* [x] offline docs cannot use a local web server — yes, it's just static HTML change.
* [x] the contents of older doc versions cannot be modified — that's unfortunate, but it does stop the problem from being perpetuated in newer releases.

kornelski commented on first-edition/book.toml:
```diff
@@ -3,4 +3,4 @@ title = "The Rust Programming Language"
 
 [output.html]
-additional-css = ["src/theme/first-edition.css"]
+additional-css = ["theme/first-edition.css"]
```
Yes. I'll test it with 0.1.

steveklabnik replied:
Thanks! To be clear, either option is totally fine; we will want to eventually update anyway. I just need to know if I have to send in another PR upstream.

kornelski replied:
I've verified it works with mdbook 0.1.7

kornelski commented:
I've changed the approach. I've removed the unnecessary upgrade to mdbook, and complex index template. 

It's now a minimal template, and specifically for mdbook 0.1.7 used by this book.

kornelski commented:
I can make another PR for mdbook upgrade if you're interested.

carols10cents commented:
> Could you also do this for the second edition as well, please? Thank you!

Triage: Looks like `second-edition` still needs to be done as Steve requested. IMO, this also should be done for `2018-edition`, so that the top level `src` is the only edition that gets indexed.

kornelski commented:
I've copied the change to `second-edition` and `2018-edition`.

kornelski commented:
Thanks @carols10cents I'll try to get it on stable ASAP. #1832
